app/controllers/api/v0/articles_controller.rb

```diff
@@ -72,11 +72,22 @@ def update
 
       def article_params
         allowed_params = [
-          :title, :body_markdown, :published, :series, :publish_under_org,
+          :title, :body_markdown, :published, :series,
           :main_image, :canonical_url, :description, tags: []
         ]
+        allowed_params << :organization_id if params["article"]["organization_id"] && allowed_to_change_org_id?
         params.require(:article).permit(allowed_params)
       end
+
+      def allowed_to_change_org_id?
+        potential_user = @article&.user || @user
+        if @article.nil? || OrganizationMembership.exists?(user: potential_user, organization_id: params["article"]["organization_id"])
+          OrganizationMembership.exists?(user: potential_user, organization_id: params["article"]["organization_id"])
```

Comment:
@Zhao-Andy I know this is already merged and stuff, I was wondering why re-running the same query twice. Maybe you can add a comment to the method explaning the logic because it's not super clear at a glance.

Thanks for the great work BTW!

Reply:
I suppose it's an opportunity for optimization :slightly_smiling_face:

Reply:
Hmm that is good question! 😅 I'll update this with the proper logic. Thanks for flagging it!